src/Domain/Block.hpp

```diff
@@ -46,10 +47,15 @@ class Block {
   /// \param id a unique ID.
   /// \param neighbors info about the Blocks that share a codimension 1
   /// boundary with this Block.
+  /// \param external_boundary_conditions the boundary conditions to be applied
+  ///        at external boundaries.
   Block(std::unique_ptr<domain::CoordinateMapBase<
             Frame::Logical, Frame::Inertial, VolumeDim>>&& stationary_map,
-        size_t id,
-        DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors) noexcept;
+        size_t id, DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors,
+        DirectionMap<
+            VolumeDim,
+            std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>
+            external_boundary_conditions = {}) noexcept;
```

Comment:
Do you need this to have a default for a particular reason? Once the consistency check in the implementation can be enabled I can't think of a case where one would not have to think about the set of boundary conditions when constructing a Block.

Reply:
Yes, once everything is updated we don't need the default, but I wasn't really feeling motivated to find all the defaults.